tools/misc/mapping_extractor.py

```
@@ -0,0 +1,259 @@
+        wb_raw = load_workbook(filename=source_file, read_only=True, data_only=True)
+    except Exception as e:
+        raise RuntimeError(f"Unable to open Excel file with openpyxl: \"{source_file}\"\n\t\t {e}")
+
+    if source_sheet not in wb_raw.sheetnames:
+        raise ValueError(f"The sheet \"{source_sheet}\" does not exist in the Excel file.")
+
+    try:
+        ws_raw = wb_raw[source_sheet]
+        header_cells = list(ws_raw.iter_rows(min_row=header_row, max_row=header_row, values_only=True))[0]
+    except Exception as e:
+        raise RuntimeError(f"Failed to read header row ({header_row}) in sheet \"{source_sheet}\":\n\t\t {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)